compatibility_server/pip_checker.py

```diff
@@ -49,6 +57,29 @@
 PIP_CHECK_CONFLICTS_PATTERN = re.compile(
     r'(.*)has requirement(.*)but you have(.*)')
 
+# docker error
+DOCKER_ERROR_MEASURE = measure_module.MeasureInt(
+    'docker_error', 'The number of docker errors.', 'Errors')
+DOCKER_ERROR_VIEW = view_module.View(
+    "docker_error_count",
+    "The number of the docker errors",
+    [],
+    DOCKER_ERROR_MEASURE,
+    aggregation_module.CountAggregation())
+
+
+def _enable_metrics(stats, view, export_to_stackdriver=False):
+    if export_to_stackdriver:
+        try:
+            _, project_id = google_auth.default()
+        except google_auth.exceptions.DefaultCredentialsError:
+            raise ValueError("Couldn't find Google Cloud credentials, set the "
+                             "project ID with 'gcloud set project'")
+        exporter = stackdriver_exporter.new_stats_exporter(
+           stackdriver_exporter.Options(project_id=project_id))
+        stats.view_manager.register_exporter(exporter)
+    stats.view_manager.register_view(view)
```

Comment:
What does this do? Maybe some docs for this function.

Reply:
done

Reply:
You are just explaining what each line of code does. What does it mean to register the view? What does registering the view do if we aren't exporting to stackdriver, etc.

Reply:
Unfortunately, my understanding of how these opencensus functions operate once they are invoked is limited since they didn't have any better documentation for Python. I updated the docstring with what I observed when I was testing and some of their Go documentation for the same functions. https://godoc.org/go.opencensus.io/stats/view#Register

> In order for data to be collected for the given view, the view needs to be
> registered with a view manager. Once a view is registered, it reports data
> to any registered exporters.
> 
> For any data to be exported to stackdriver, an exporter needs to be created
> and registered with the view manager. Collected data will be reported via
> all the registered exporters. By not creating and registering an exporter,
> all collected data will stay local and will not appear on stackdriver.

Reply:
Maybe it makes sense to move the last line (` stats.view_manager.register_view(view)` in this method somewhere else. I looks like `register_view` is idempotent (https://github.com/census-instrumentation/opencensus-python/blob/dfae33d14d48a7a23ef25b8ac886acfcccfe3ee7/opencensus/stats/view_manager.py) but I'd add a comment pointing that out if you want to call it on every pip check.

Then you could remove the `if export_to_stackdriver` and call this:
```python
def _export_to_stackdriver(stats: stats_module.Stats):
    """Exports the given statistics to Stackdriver.

    In order for data to be collected for the given view, the view needs to be
    registered with a view manager. Once a view is registered, it reports data
    to any registered exporters.

    For any data to be exported to stackdriver, an exporter needs to be created
    and registered with the view manager. Collected data will be reported via
    all the registered exporters. By not creating and registering an exporter,
    all collected data will stay local and will not appear on stackdriver.

    Args:
      stats: The statistics that should be exported to Stackdriver.
    """

Reply:
done